File: src/main/java/it/alian/gun/mesmerize/compat/hook/MesmerizePlaceholder.java
package it.alian.gun.mesmerize.compat.hook;

import it.alian.gun.mesmerize.MConfig;
import it.alian.gun.mesmerize.lore.LoreInfo;
import it.alian.gun.mesmerize.lore.LoreParser;
import me.clip.placeholderapi.external.EZPlaceholderHook;
import org.bukkit.entity.Player;
import org.bukkit.plugin.Plugin;

import java.lang.reflect.Field;
import java.text.DecimalFormat;

public class MesmerizePlaceholder extends EZPlaceholderHook {

    private static final DecimalFormat format = new DecimalFormat(MConfig.Misc.customDecimalFormat);

    public MesmerizePlaceholder(Plugin plugin, String identifier) {
        super(plugin, identifier);
    }

    @Override
    public String onPlaceholderRequest(Player player, String s) {
        if (s.startsWith("stats_")) {
            String key = s.substring("stats_".length());
            LoreInfo info = LoreParser.getByEntityId(player.getEntityId());
            try {
                Field field = info.getClass().getDeclaredField(key);
                field.setAccessible(true);
                return format.format(field.get(info));
            } catch (Exception e) {
                return null;
            }
        }
        return null;
    }

}

File: src/main/scala/it/alian/gun/mesmerize/compat/hook/MesmerizePlaceholder.scala
package it.alian.gun.mesmerize.compat.hook

import java.text.DecimalFormat

import it.alian.gun.mesmerize.lore.LoreParser
import it.alian.gun.mesmerize.scalaapi.Prelude._
import me.clip.placeholderapi.external.EZPlaceholderHook
import org.bukkit.entity.Player
import org.bukkit.plugin.Plugin

object MesmerizePlaceholder {
  private val format = new DecimalFormat(config("misc.customDecimalFormat", "0.00"))
}

class MesmerizePlaceholder(val plugin: Plugin, val identifier: String) extends EZPlaceholderHook(plugin, identifier) {
  override def onPlaceholderRequest(player: Player, s: String): String =
    if (s.startsWith("stats_")) {
      val name = s.substring("stats_".length)
      val info = LoreParser.parse(player)
      config(s"prefix.$name.type", "number") match {
        case "number" => MesmerizePlaceholder.format.format(info.num(name))
        case "string" => info.str(name)
      }
    } else ""

}